docs/call-tree.md

```diff
@@ -0,0 +1,353 @@
+
+```
+    onLoad
+      ↓
+      a
+      ↓
+      b
+```
+
+# Call trees as stacks of functions (FuncStacks), and referring to nodes with function paths
+
+What we really care about in the call tree is the execution time of a given function, and its relationship to other functions. Since there can be multiple frames and stacks per function in a call tree, the logical step is to combine all of these stacks and frames together in terms of what function they point to. This combined representation is known in perf.html as the funcStack.
+
+However once this happens, it starts to become quite difficult to explain relationships and refer to specific nodes within this modified view. Whenever we apply any type of data transformation, these funcStacks need to be regenerated, and the indexes will be different. The only unique identifier that exists is the function, and where it exists in the tree. So in order refer to a specific funcStack, we need a different way to store a reference to the tree. This structure in perf.html is called the FuncPath.
+
+```
+              A
+              |
+              v
+              B
+            /   \
+           v     v
+          C       H
+        /   \      \
+       v     v      v
+      D       F     I
+      |       |
+      v       v
+      E       G
+```
+
+Given the above call tree, the IDs of the nodes are rather random, and will change as we further modify the structure of the call tree. So to refer to the node at function C it is useful to provide a path of functions that shows how to get there. In our case this would be the list of functions `A ➡ B ➡ C`. The `FuncPath` in perf.html would then be stored as an array of function indexes.
+
+# Modifying the tree
+
+Call trees are interesting for the information they provides, but they can be quite large structures to navigate. In order to provide a proper analysis we need to be able to modify the structure and shape of the tree to provide more intelligent results. There is not a great body of language available to describe these actions, so they are listed here below.
+
+## Again looking at these samples
+
+```
+  A    A    A
+  |    |    |
+  v    v    v
+  B    B    B
+  |    |    |
+  v    v    v
+  C    C    H
+  |    |    |
+  v    v    v
+  D    F    I
+  |    |
+  v    v
+  E    G
+```
+
+## Would produce this call tree
+
+```
+.                  A:3,0            
+.                    |              
+.                    v              
+.                  B:3,0            
+.                  /    \           
+.                 v      v          
+.             C:2,0     H:1,0       
+.            /      \         \     
+.           v        v         v    
+.         D:1,0     F:1,0     I:1,1
+.         |           |             
+.         v           v             
+.       E:1,1       G:1,1           
```

Comment:
Thanks, I did that to make it easy to copy/paste in my editor to avoid auto-indenting behavior. I didn't clean it up after I was done.